admin_guide/enabling_EFK.adoc

```diff
@@ -0,0 +1,357 @@
+
+=== ElasticSearch
+
+All pods in OpenShift deployments can safely share any volumes specified in the
+deployment. With ElasticSearch, all pods must have their own storage.
+
+The ability to specify multiple volumes to be allocated to each instance in a
+deployment is currently being completed. Currently, multiple deployments are
+used in order to scale ElasticSearch. To view all current deployments used by
+ElasticSearch:
+
+====
+----
+$ oc get dc --selector logging-infra=elasticsearch
+----
+====
+
+To scale ElasticSearch deployments, create and add more deployments, being aware
+of the cluster parameter restrictions. The deployer uses a template to create
+ElasticSearch deployments. These deployments will be named differently, but will
+all have the 'service/logging-es-cluster' prefix:
+
+====
+----
+$ oc process logging-es-template | oc create -f -
+----
+====
+
+*ElasticSearch Storage*
+
+The deployer creates a ephemeral deployment in which all of a pod's data is lost
+upon restart. For production,
+link:../admin_guide/persistent_storage_nfs.html[specify a persistent storage
+volume] for each ElasticSearch deployment. You can use the 'oc volume' command
+to add a volume to a deployment:
+
+====
+----
+$ oc volume dc/logging-es-rca2m9u8 \
+          --add --overwrite --name=elasticsearch-storage \
+          --type=hostPath --path=/path/to/storage
+----
+====
+
+*Node Selector*
+
+As ElasticSearch can use a lot resources, all members of a cluster should have
+low latency network connections to each other. Do this by directing the
+instances to dedicated nodes, or a dedicated region within your cluster, using a
+node selector.
+
+To configure a node selector, edit each deployment configuration and add the
+'nodeSelector' parameter to specify the label of the desired nodes:
+
+====
+----
+apiVersion: v1
+kind: DeploymentConfig
+spec:
+  nodeSelector:
+    nodelabel: logging-es-node-1
+----
+====
+
+=== Fluentd
+
+Once ElasticSearch is running, scale Fluentd to every node to feed logs into
+ElasticSearch:
+
+====
+----
+$ oc scale dc/logging-fluentd --replicas=3
+$ oc scale rc/logging-fluentd-1 --replicas=3
+----
+====
+
+=== Kibana
+
+To access the Kibana console from the OpenShift web console, add the
+`loggingPublicURL` parameter to the *_/etc/openshift/master/master-config.yaml_*
+file, with the URL of the Kibana console. The value must be a HTTPS URL:
+
+====
+----
+...
+assetConfig:
+  ...
+  loggingPublicURL: "https://<kibana-hostname>"
+...
+----
+====
+
+Setting the `loggingPublicURL` parameter creates a *View Archive* button on the
+OpenShift web console under the *Browse* tab, then finding the desired pod. This
+links to the Kibana console.
+
+You can scale the Kibana deployment as usual for redundancy:
+
+====
+----
+$ oc scale dc/logging-kibana --replicas=2
+$ oc scale rc/logging-kibana-1 --replicas=2
+----
+====
+
+You can see the UI by visiting the site specified at the '*KIBANA_HOSTNAME*'
+variable.
```

Comment:
... assuming you have a router and DNS points at it appropriately for this address. Which we should probably mention much earlier.